crates/ark/src/lsp/completions/sources/composite/pipe.rs

```diff
@@ -124,69 +121,51 @@ fn eval_pipe_root(name: &str) -> Option<RObject> {
     Some(value)
 }
 
-fn find_pipe_root_name(context: &DocumentContext, node: &Node) -> Option<String> {
-    // Try to figure out the code associated with the 'root' of the pipe expression.
-    let root = local! {
-
-        let root = find_pipe_root_node(context, *node)?;
-        is_pipe_operator(context, &root).into_option()?;
+fn find_pipe_root_name(context: &DocumentContext, node: &Node) -> anyhow::Result<Option<String>> {
```

Comment:
I was prompted to clean this up by the fact that we now propagate up the `Result`, and I think it cleaned up nicely.